5.Enhancing the app/controllers/admin.js

```diff
@@ -0,0 +1,64 @@
+
+const Product = require('../models/product');
+const Cart = require('../models/cart');
+
+const ApiError = require('../error/ApiError');
+
+//For Add-Product Page
+exports.getAddProduct = (req,res,next) => {
+    res.render('admin/add-product',{isLoggedIn:req.session.isLoggedIn});
+};
+
+exports.postAddProduct = (req,res,next) => {
+    res.redirect('/products');
+    if(req.body.productName == "") {
+        next( ApiError.badRequest('Product title is necessary'));
+        return;
+    }
+    let product = new Product(req.body.productName,req.body.imageUrl,req.body.price,req.body.description);
+    product.save();
+}
+
+//For getting all the products on Admin-side
+exports.getAdminProducts = (req,res,next) => {
+    let products = Product.fetchAll();
+    res.render('admin/products',{prods:products, title:'Admin Products'});
+}
+
+//To Update the product
+exports.updateProduct = (req,res,next) => {
+    const prodId = req.body.productId;
```

Comment:
use destructuring to get productId from the body object. in that case, you don't need to rename the variable.